Improve exception handling with InvocationTargetException

Currently some exceptions are catched if the invoked method fails but I see only
`2022-01-05 23:36:04.013 [ERROR] [on.jrule.internal.engine.JRuleEngine] - [UpdateWeightRule] Error` in the openhab log. Which is not helpful.
https://github.com/seaside1/jrule/blob/39b467cbdac0b754061d11f09b532a984c96557c/src/main/java/org/openhab/automation/jrule/internal/engine/JRuleEngine.java#L467-L471
I guess there is a [InvocationTargetException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/InvocationTargetException.html) caused in some internal classes that are called by a rule of mine. 
In case of an `InvocationTargetException` the actual cause of the exception has to be unwrapped via `getCause()`. Please see linked javadoc as reference.
Can you please add something like:
```java
catch (InvocationTargetException ite) {
    JRuleLog.error(logger, context.getRuleName(), "Error", ite.getCause()); 
}
```
or maybe `"Error {}"` also helps.

--- a/src/main/java/org/openhab/automation/jrule/internal/engine/JRuleEngine.java
+++ b/src/main/java/org/openhab/automation/jrule/internal/engine/JRuleEngine.java
@@ -14,6 +14,9 @@ package org.openhab.automation.jrule.internal.engine;
 
 import java.beans.PropertyChangeEvent;
 import java.beans.PropertyChangeListener;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
@@ -466,8 +469,21 @@ public class JRuleEngine implements PropertyChangeListener {
         rule.setRuleLogName(context.getLogName());
         try {
             final Object invoke = context.isEventParameterPresent() ? method.invoke(rule, event) : method.invoke(rule);
-        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException | SecurityException e) {
-            JRuleLog.error(logger, context.getRuleName(), "Error", e);
+        } catch (IllegalAccessException | IllegalArgumentException | SecurityException e) {
+            JRuleLog.error(logger, context.getRuleName(), "Error {}", e);
+        } catch (InvocationTargetException e) {
+            Throwable ex = e.getCause() != null ? e.getCause() : null;
+            JRuleLog.error(logger, context.getRuleName(), "Error message: {}", ex.getMessage());
+            JRuleLog.error(logger, context.getRuleName(), "Error Stacktrace: {}", getStackTraceAsString(ex));
+        }
+    }
+
+    private synchronized static String getStackTraceAsString(Throwable throwable) {
+        try (StringWriter sw = new StringWriter(); PrintWriter pw = new PrintWriter(sw)) {
+            throwable.printStackTrace(pw);
+            return sw.toString();
+        } catch (IOException ioe) {
+            throw new IllegalStateException(ioe);
         }
     }
 }

